services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -3683,14 +3683,27 @@ private OfflinePushStatusInfo getOffLineJobStatus(
                 currentReturnStatus,
                 currentReturnStatusDetails);
           }
-          // status PUSHED is set when batch store's target region push is completed, but other region are yet to
-          // complete
-          if (isTargetRegionPush && !isVersionPushed) {
-            parentStore.updateVersionStatus(versionNum, PUSHED);
-            repository.updateStore(parentStore);
-          } else { // status ONLINE is set when all region finishes ingestion for either regular or target region push.
+
+          /**
+           * Check if the latest version is online and is completed in all regions.
+           * If it is online in all regions, set parent version status to ONLINE
+           * Otherwise, set parent version status to PUSHED
+           */
+          Map<String, Integer> currentVersionsToColo = getCurrentVersionsForMultiColos(clusterName, storeName);
+          int latestVersionNum = parentStore.getLargestUsedVersionNumber();
+          boolean isLatestVersionOnlineInAllRegions = true;
+          for (Map.Entry<String, Integer> entry: currentVersionsToColo.entrySet()) {
+            if (currentVersionsToColo.get(entry.getKey()) != latestVersionNum) {
```

Comment:
This assumption is not correct, the versions in parent need not be same as in child.